infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/SshKeysProvisioner.java

```diff
@@ -113,20 +143,41 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
                 sshManager.generatePair(
                     identity.getOwnerId(), "vcs", "default-" + new Date().getTime()));
       } catch (ServerException | ConflictException e) {
-        LOG.warn("Unable to generate the initial SSH key. Cause: {}", e.getMessage());
-        return;
+        String message =
+            format("Unable to generate the initial SSH key. Cause: %s", e.getMessage());
+        LOG.warn(message);
+        k8sEn.addWarning(
+            new WarningImpl(
+                NOT_ABLE_TO_PROVISION_SSH_KEYS,
+                format(NOT_ABLE_TO_PROVISION_SSH_KEYS_MESSAGE, message)));
+        throw new InfrastructureException(e);
       }
     }
+    return sshPairs;
+  }
 
-    doProvisionSshKeys(sshPairs, k8sEnv, identity.getWorkspaceId());
-
-    StringBuilder sshConfigData = new StringBuilder();
-    for (SshPair sshPair : sshPairs) {
-      sshConfigData.append(buildConfig(sshPair.getName()));
+  /**
+   * Return system (aka 'internal') ssh keys, this key used for internal services like rsync via ssh
+   * and etc. Not generated and managed by a user
```

Comment:
This doc line could be clearer, one reading is that they're not generated and instead are managed by a user. Maybe `Not generated or managed by a user`?

Reply:
Good catch, thanks